src/back-end/test/deleteUser.test.js

```diff
@@ -0,0 +1,80 @@
+/**
+ * Tests export user deletion functions
+ */
+
+/* Mock envvars */
+process.env.HASHKEY = "AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA";
+
+/* Imports */
+const { createUser } = require("../createFiles/createUser");
+const { deleteUser } = require("../deleteFiles/deleteUser");
+const schema = require("../schema.js");
+const mongoose = require("mongoose");
+
+describe("deleteUser Tests", () => {
+
+    /**
+     * Checks for an existing user
+     *
+     * @param {Object} userData Object containing email and password
+     * @return true if user exists, false otherwise
+     */
+     const checkUser = async (userData) => {
+        try {
+            const user = await schema.User.findOne({ email: userData.email});
+            if (user === null) {
+                return false;
+            }
+        } catch (err) {
+            return false;
+        }
+    };
+
+    /* Connect to the in-memory Mongo server */
+    beforeAll(async () => {
+        mongoose.set("useCreateIndex", true);
+        await mongoose.connect(`${globalThis.__MONGO_URI__}${globalThis.__MONGO_DB_NAME__}`, {
+            useUnifiedTopology: true,
+            useNewUrlParser: true
+        });
+    });
+
+    /* Drop the database */
+    afterEach(async () => {
+        await schema.User.deleteMany({});
+    });
+
+    /* Clean up the connection */
+    afterAll(async () => {
+        await mongoose.connection.close();
+    });
+
+    test("User is properly deleted", async () => {
+        const EMAIL = "foo@gmail.com";
+        const PWDHASH = "123ABC";
+        const KEY = "ABC123";
+        const user = {
+            email: EMAIL,
+            pwd: PWDHASH
+        };
+        await createUser(EMAIL, PWDHASH, KEY, () => {
+            let success = checkUser(user);
+            expect(success).toBe(true);
+        });
+        await deleteUser(EMAIL, () => {
+            let fail = checkUser(user);
+            expect(fail).toBe(false);
+        })
+    });
+
+    test("Non-existing user cannot be deleted", async () => {
+        const EMAIL = "foo@gmail.com";
+        await deleteUser(EMAIL).
```

Comment:
If you're going to use `then` and `catch` you need to have just pass in `done` and call it at the end of `then` and `catch`. No need for `await` or for this function to be marked `async`.